synthesizer/process/src/cost.rs

```diff
@@ -460,6 +465,29 @@ pub fn cost_in_microcredits_v1<N: Network>(stack: &Stack<N>, function_name: &Ide
         })
 }
 
+/// Returns the compute cost for a transaction in microcredits.
+/// This is used to limit the amount of compute in the block generation hot
+/// path. This does NOT represent the full costs which a user has to pay.
+pub fn compute_cost<N: Network>(process: &Process<N>, transaction: &Transaction<N>) -> Result<u64> {
```

Comment:
Could we call `compute_cost` from `deployment_cost` and `execution_cost_v*`?

Reply:
Afaict, neither of those functions have access to all the inputs required for `compute_cost`? While I'm all for code reuse, I'm not sure the extra indirection and type-wrangling would be worth it here. But maybe I'm missing something, happy to take a closer look 🕵️‍♂️

Reply:
```
pub fn deployment_cost<N: Network>(deployment: &Deployment<N>)
pub fn execution_cost_v2<N: Network>(process: &Process<N>, execution: &Execution<N>)
```

You can create a Transaction from the Deployment or Execution. If that's too ugly we could have a separate `deployment_compute_cost` and `execute_compute_cost`.

Reply:
Yeah, it's the extra fields (various ids, the fee) that make this cumbersome but I think that second option would be fine 👍

Reply:
I added `execution_compute_cost` in [f3393c0](https://github.com/ProvableHQ/snarkVM/pull/2601/commits/f3393c088f9e2506935b3aed6e22a0417253011b), hopefully this is what you had in mind? There was already a `deployment_synthesis_cost` so I left that one as is for now.